src/service/exception_handlers.py

```diff
@@ -0,0 +1,82 @@
+"""
+Exception handlers for the FastAPI application.
+"""
+
+import logging
+
+from fastapi import HTTPException, Request, status
+from fastapi.exceptions import RequestValidationError
+from fastapi.responses import JSONResponse
+
+from src.service import errors
+from src.service.error_mapping import map_error
+from src.service.exceptions import SparkManagerError
+
+logger = logging.getLogger(__name__)
+
+
+def _format_error(
+    status_code: int,
+    error_code: int | None,
+    error_type_str: str | None,
+    message: str | None,
+):
+    """Format error response with consistent structure."""
+    return JSONResponse(
+        status_code=status_code,
+        content={
+            "error": error_code,
+            "error_type": error_type_str,
+            "message": message or error_type_str or "Unknown error",
+        },
+    )
+
+
+async def universal_error_handler(request: Request, exc: Exception):
+    """
+    Universal handler for all types of exceptions.
+
+    Handles:
+    - SparkManagerError and its subclasses (including all specific error types)
+    - HTTPException from FastAPI
+    - RequestValidationError for request validation
+    - Generic exceptions
+    """
+    # Default values
+    error_code = None
+    error_type_str = None
+    status_code = status.HTTP_500_INTERNAL_SERVER_ERROR
+    message = None
+
+    if isinstance(exc, SparkManagerError):
+        # handle SparkManagerError and subclasses
+        error_type, status_code = map_error(exc)
+
+        # Extract values from error_type if available
+        if error_type:
+            error_code = error_type.error_code
+            error_type_str = error_type.error_type
+        else:
+            raise ValueError("this should be impossible")
+
+        # Get the exception message, falling back to the error type string
+        message = str(exc) if str(exc) else error_type_str
+
+    elif isinstance(exc, RequestValidationError):
```

Comment:
Does this work? From before it looked like this wasn't getting passed to the error handler at all

Reply:
Yea. It works. I wasn't formatting the message. I think `str(exc.errors())` should be fine. 

```
(dev) tgu@tgu $curl -X 'POST' \
  'http://127.0.0.1:8000/clusters' \
  -H 'accept: application/json' \
  -H 'Authorization: Bearer FOUWOXFQMZDR5TYBQLP5YUEAFQDEI7OR' \
  -H 'Content-Type: application/json' \
  -d '{
  "worker_count": 2,
  "worker_cores": 1,
  "worker_memory": "1",
  "master_cores": 1,
  "master_memory": "1G"
}'
{"error":30010,"error_type":"Request validation failed","message":"[{'type': 'value_error', 'loc': ('body', 'worker_memory'), 'msg': \"Value error, Memory must be in format '{int}G', got '1'\", 'input': '1', 'ctx': {'error': ValueError(\"Memory must be in format '{int}G', got '1'\")}}]"}%   
```